### PR TITLE
Create config overrides directory if needed

### LINKED ISSUE #26: Open Liberty 0.2.0 apps fail to start up because of missing directory
## What happened?

The 0.2.0 release fails to run because of `<runtime-root>/usr/servers/defaultServer/configDropins/overrides` directory does not exist.

* What were you attempting to do? Running an OL app built using 0.2.0 release.

* What did you expect to happen? Starting an OL app to succeed.

* What was the actual behavior? Please provide log output, if possible. Build failed with the error `open /cnb/buildpacks/paketo-buildpacks_open-liberty/0.2.0/templates: no such file or directory`

```text
Setting Active Processor Count to 4
Calculating JVM memory based on 1415100K available memory
Calculated JVM Memory Configuration: -XX:MaxDirectMemorySize=10M -Xmx650999K -XX:MaxMetaspaceSize=252100K -XX:ReservedCodeCacheSize=240M -Xss1M (Total Memory: 1415100K, Thread Count: 250, Loaded Class Count: 42095, Headroom: 0%)
Enabling Java Native Memory Tracking
Adding 128 container CA certificates to JVM truststore
unable to configure
unable to contribute app and config to runtime root: unable to create file '{/layers/paketo-buildpacks_open-liberty/open-liberty-runtime-full/usr/servers/defaultServer/apps/app / war}':
open /layers/paketo-buildpacks_open-liberty/open-liberty-runtime-full/usr/servers/defaultServer/configDropins/overrides/app.xml: no such file or directory
ERROR: failed to launch: exec.d: failed to execute exec.d file at path '/layers/paketo-buildpacks_open-liberty/helper/exec.d/linker': exit status 1
```

## Build Configuration

* What platform (`pack`, `kpack`, `tekton` buildpacks plugin, etc.) are you
  using? Please include a version. Pack version 0.23.0-rc1+git-4d5680a.build-3019

* What buildpacks are you using? Please include versions. paketo-buildpacks/open-liberty:0.2.0

* What builder are you using? If custom, can you provide the output from `pack
  inspect-builder <builder>`? base

* Can you provide a sample app or relevant configuration (`buildpack.yml`,
  `nginx.conf`, etc.)? Running any Open Liberty app fails.

## Checklist

* [X] I have included log output.
* [X] The log output includes an error message.
* [X] I have included steps for reproduction.

### PR DESCRIPTION
**Pull request by hibell**
Fixes #26.

## Summary

This change creates the <runtime-root>/usr/servers/defaultServer/configDropins/overrides directory if needed.

## Use Cases

Fix an issue where OL 0.2.0 apps do not start up because of missing config overrides directory.

## Checklist

* [X] I have viewed, signed, and submitted the Contributor License Agreement.
* [X] I have linked issue(s) that this PR should close using keywords or the Github UI (See [docs](https://docs.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue))
* [X] I have added an integration test, if necessary.
* [X] I have reviewed the [styleguide](https://github.com/paketo-buildpacks/community/blob/main/STYLEGUIDE.md) for guidance on my code quality.
* [X] I'm happy with the commit history on this PR (I have rebased/squashed as needed).

### CODE REVIEW

File: helper/linker.go [hunk 1]
```diff
@@ -62,6 +62,10 @@ func (f FileLinker) Execute() (map[string]string, error) {
 		layerDir = "/layers/paketo-buildpacks_open-liberty/open-liberty-runtime"
 	}
 
+	if _, err := os.Stat(layerDir); os.IsNotExist(err) {
+		return map[string]string{}, fmt.Errorf("runtime root '%v' does not exist", layerDir)
+	}
+
```
Comment (dmikusa-pivotal) [inline helper/linker.go:68]:
```suggestion
        _, err := os.Stat(layerDir)
	if err != nil && os.IsNotExist(err) {
		return map[string]string{}, fmt.Errorf("unable to find '%s', folder does not exist", layerDir)
	} else if err != nil {
	        return map[string]string{}, fmt.Errorf("unable to check %s\n%w", layerDir, err)
	}
```

I think you're missing an error check here.

File: helper/linker_test.go [hunk 2]
```diff
@@ -72,8 +72,6 @@ func testLink(t *testing.T, context spec.G, it spec.S) {
 			Expect(os.Setenv("BPI_OL_BASE_ROOT", baseLayerDir)).To(Succeed())
 
 			Expect(os.MkdirAll(filepath.Join(layerDir, "usr", "servers", "defaultServer", "apps"), 0755)).To(Succeed())
-			Expect(os.MkdirAll(filepath.Join(layerDir, "usr", "servers", "defaultServer", "configDropins", "overrides"), 0755)).To(Succeed())
```
Comment (dmikusa-pivotal) [inline helper/linker_test.go:75]: Sorry, I'm not seeing it. What's the need to remove these lines? We're putting things into `layerDir/usr/servers/defaultServer/configDropins/overrides`, shouldn't we be cleaning that up?

File: helper/linker_test.go [hunk 3]
```diff
@@ -87,7 +85,6 @@ func testLink(t *testing.T, context spec.G, it spec.S) {
 			Expect(os.Unsetenv("BPI_OL_BASE_ROOT")).To(Succeed())
 
 			Expect(os.RemoveAll(filepath.Join(layerDir, "usr", "servers", "defaultServer", "apps"))).To(Succeed())
-			Expect(os.RemoveAll(filepath.Join(layerDir, "usr", "servers", "defaultServer", "configDropins", "overrides"))).To(Succeed())
```
Comment (dmikusa-pivotal) [inline helper/linker_test.go:90]: Same as previous comment.

**Review by dmikusa-pivotal — CHANGES_REQUESTED** [review]

File: helper/linker_test.go (hunk 2 above)
Comment (hibell) [reply helper/linker_test.go]: I wanted to check that the linker was creating the config overrides directory if it doesn't exist instead. I think adding a new test might be the best approach for that though. Thanks.

File: helper/linker.go [hunk 4]
```diff
@@ -62,6 +62,12 @@ func (f FileLinker) Execute() (map[string]string, error) {
 		layerDir = "/layers/paketo-buildpacks_open-liberty/open-liberty-runtime"
 	}
 
+	_, err := os.Stat(layerDir)
```
Comment (dmikusa-pivotal) [inline helper/linker.go:65]:
```suggestion
	_, err = os.Stat(layerDir)
```

**Review by dmikusa-pivotal — APPROVED** [review]

**Pull request merged**